docs/DeveloperGuide.md

```diff
@@ -0,0 +1,123 @@
+# RepoSense - Developer Guide
+Thank you for your interest in contributing to RepoSense!
+- [Setting up](#setting-up)
+  - [Prerequisites](#prerequisites)
+  - [Setting up the project in your computer using IntelliJ](#setting-up-the-project-in-your-computer-using-intellij)
+  - [Verifying the setup](#verifying-the-setup)
+  - [Configuring the coding style](#configuring-the-coding-style)
+- [Architecture](#architecture)
+  - [Parser](#parser)
+  - [Git](#git)
+  - [CommitsReporter](#commitsreporter)
+  - [AuthorshipReporter](#authorshipreporter)
+  - [ReportGenerator](#reportgenerator)
+  - [System](#system)
+  - [Model](#model)
+- [HTML Dashboard](#html-dashboard)
+
+## Setting up
+
+### Prerequisites
+1. **JDK `1.8.0_60`** or later.
+2. **git** on the command line.
+ > Check that the tools exist on your OS terminal by typing its name on the terminal and ensure that the terminal does not output messages such as `not found` or `not recognized`.
```

Comment:
This line is not relevant anymore?

Reply:
>This line is not relevant anymore?

The line is still relevant for the dependency on **git**, which I don't think is available by default (for Windows at least).

Reply:
True. rephrase it to be specific to git?

Reply:
Noted, will update to to specify version `2.14` or later to resolve #127 at the same time.